test/byContract/IexecAccessors/IexecAccessors.test.ts

```diff
@@ -102,4 +117,122 @@ describe('IexecAccessors', async () => {
+                workerpool: workerpoolAddress,
+            };
+        });
+        it('Should get result of task', async function () {
+            const oracleConsumerInstance = await new TestClient__factory()
+                .connect(anyone)
+                .deploy()
+                .then((contract) => contract.deployed());
+            const orders = buildOrders({
+                assets: ordersAssets,
+                requester: requester.address,
+                callback: oracleConsumerInstance.address,
+            });
+            const { dealId, taskId, taskIndex } = await iexecWrapper.signAndMatchOrders(
+                ...orders.toArray(),
+            );
+
+            await iexecPocoAsAnyone.initialize(dealId, taskIndex).then((tx) => tx.wait());
+
+            const workerTaskStake = await iexecPocoAsAnyone
+                .viewDeal(dealId)
+                .then((deal) => deal.workerStake.toNumber());
+            const { resultHash, resultSeal } = buildResultHashAndResultSeal(
+                taskId,
+                callbackResultDigest,
+                worker1,
+            );
+            const schedulerSignature = await buildAndSignContributionAuthorizationMessage(
+                worker1.address,
+                taskId,
+                AddressZero,
+                scheduler,
+            );
+            await iexecWrapper.depositInIexecAccount(worker1, workerTaskStake);
+            await iexecPocoAsAnyone
+                .connect(worker1)
+                .contribute(taskId, resultHash, resultSeal, AddressZero, '0x', schedulerSignature)
+                .then((tx) => tx.wait());
+
+            await iexecPocoAsAnyone
+                .connect(worker1)
+                .reveal(taskId, callbackResultDigest)
+                .then((tx) => tx.wait());
+            await iexecPocoAsAnyone
+                .connect(scheduler)
+                .finalize(taskId, results, resultsCallback)
+                .then((tx) => tx.wait());
+            const task = await iexecPocoAsAnyone.viewTask(taskId);
+            await expect(task.status).to.equal(TaskStatusEnum.COMPLETED);
+            await expect(await iexecPocoAsAnyone.callStatic.resultFor(taskId)).to.equal(
+                resultsCallback,
+            );
+        });
+        it('Should not get result when task is not completed', async function () {
+            const volume = 3;
+            const orders = buildOrders({
+                assets: ordersAssets,
+                requester: requester.address,
+                volume,
+            });
+
+            const { dealId } = await iexecWrapper.signAndMatchOrders(...orders.toArray());
+
+            const tasks: string[] = [];
+            tasks.push(getTaskId(dealId, 0));
+            for (let taskIndex = 1; taskIndex < volume; taskIndex++) {
+                await iexecPocoAsAnyone.initialize(dealId, taskIndex).then((tx) => tx.wait());
+                tasks.push(getTaskId(dealId, taskIndex));
+            }
```

Comment:
Since we only have 3 tasks I think it would be more readable to do something like:
```ts
const unsetTaskId = getTaskId(dealId, 0);
const activeTaskId = await initializeTask(dealId, 1);
const revealingTaskId = await initiliazeTask(dealId, 2).then(taskId => contribute(taskId);

await verifyTaskStatusAndResult(unsetTaskId, TaskStatusEnum.UNSET);
await verifyTaskStatusAndResult(activeTaskId, TaskStatusEnum.ACTIVE);
await verifyTaskStatusAndResult(revealingTaskId, TaskStatusEnum.REVEALING);
```
and you can re-use `initializeTask()` and `contribute()` in the first test.